### PR TITLE
Implement getSearchProvider

### PR DESCRIPTION
**Pull request by LEDfan**
@tanghus Is this the correct way? I'll add the Docs in a minute.

### CODE REVIEW

**Comment by LEDfan** [comment]
@tanghus can you review this please?

File: appinfo/app.php [hunk 1]
```diff
@@ -64,7 +64,8 @@
 	$addressBooks = $app->getAddressBooksForUser();
 	foreach ($addressBooks as $addressBook)  {
 		if ($addressBook->isActive()) {
-			\OCP\Contacts::registerAddressBook(new AddressbookProvider($addressBook));
+			// $addressBook is instance of \OCA\Contacts\Addressbook
+			\OCP\Contacts::registerAddressBook($addressBook->getSearchProvider($addressBook));
```
Comment (tanghus) [inline appinfo/app.php]: This should just be: `\OCP\Contacts::registerAddressBook($addressBook->getSearchProvider());`

File: lib/addressbook.php [hunk 2]
```diff
@@ -482,4 +482,12 @@ public function getBirthdayEvents() {
...
 		return $events;
 	}
+
+	/**
+	 * Returns the searchProvider for a specific backend.
+	 * @param \OCA\Contacts\Addressbook $addressbook
+	 */
+	public function getSearchProvider($addressbook){
+		return $this->backend->getSearchProvider($addressbook);
+	}
```
Comment (tanghus) [inline lib/addressbook.php:492]: And this should be 

``` php
    public function getSearchProvider() {
        return $this->backend->getSearchProvider($this);
    }
```

**Comment by tanghus** [comment]
Sorry for the delay. I was away for the Easter.

**Comment by LEDfan** [comment]
@tanghus solved all issues :)

File: lib/addressbook.php [hunk 3]
```diff
@@ -482,4 +482,12 @@ public function getBirthdayEvents() {
 
 		return $events;
 	}
+
+	/**
+	 * Returns the searchProvider for a specific backend.
+	 * @param \OCA\Contacts\Addressbook $addressbook
```
Comment (tanghus) [inline lib/addressbook.php:488]: Guess this is superfluous ;)

**Comment by tanghus** [comment]
Thanks a lot @LEDfan

**Pull request merged**